Windows: replace POSIX path-separator assertions in tests (`/` vs `\\`)

## Background

Tests assert against path strings using POSIX \`/\` separators, breaking on Windows where \`str(Path(...))\` produces \`\\\` separators.

Surfaces in PR #638's first Windows matrix run.

## Symptoms

\`\`\`
FAILED test_wiki_cmd_override.py::test_wiki_skill_override_invokes_editor
  assert False
  + where False = '...\\\\skills\\\\hello\\\\overrides\\\\claude.md'.endswith('/skills/hello/overrides/claude.md')

FAILED test_wiki_cmd_override.py::test_wiki_skill_override_stdout_contract
  assert 'Seeded skills/hello/overrides/codex.md' in 'Seeded skills\\\\hello\\\\overrides\\\\codex.md\\n...'
\`\`\`

## Affected files (8+ tests known)

- \`packages/memtomem/tests/test_wiki_cmd_override.py\` — 8 failures (all endswith / substring of POSIX paths)
- Likely scattered in: \`test_uninstall_cmd.py\`, \`test_web_routes.py\`, \`test_init_cmd.py\` (subset of their failures)

## Fix pattern

Three options per call site:
- \`endswith(os.path.join(\"skills\", ...))\` — uses native sep
- \`Path(actual).as_posix().endswith(\"/...\")\` — normalize actual to POSIX
- \`actual.replace(os.sep, \"/\")\` then compare — cheapest at scale

\`as_posix()\` is usually cleanest for assertion targets that already exist as Paths. Production code (CLI stdout) should be audited too — if \`mm wiki skill override\` prints \`Seeded skills\\hello\\overrides\\codex.md\` on Windows, that's arguably a UX fix rather than a test-only fix.

## Refs

- #634 — Windows + macOS CI matrix expansion (umbrella)
- PR #638 — matrix infrastructure (Windows currently informational)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/packages/memtomem/tests/test_config_overrides.py
+++ b/packages/memtomem/tests/test_config_overrides.py
@@ -84,7 +84,7 @@ def test_env_var_wins_over_config_json_scalar(
     )
     cfg = Mem2MemConfig()
     load_config_overrides(cfg)
-    assert str(cfg.storage.sqlite_path) == "/from/env.db"
+    assert Path(cfg.storage.sqlite_path).as_posix() == "/from/env.db"
 
 
 def test_env_var_wins_over_config_json_list(
@@ -97,7 +97,7 @@ def test_env_var_wins_over_config_json_list(
     )
     cfg = Mem2MemConfig()
     load_config_overrides(cfg)
-    assert [str(p) for p in cfg.indexing.memory_dirs] == ["/from/env"]
+    assert [Path(p).as_posix() for p in cfg.indexing.memory_dirs] == ["/from/env"]
 
 
 def test_env_and_config_coexist_on_different_fields(
@@ -117,7 +117,7 @@ def test_env_and_config_coexist_on_different_fields(
     )
     cfg = Mem2MemConfig()
     load_config_overrides(cfg)
-    assert str(cfg.storage.sqlite_path) == "/from/env.db"
+    assert Path(cfg.storage.sqlite_path).as_posix() == "/from/env.db"
     assert cfg.embedding.model == "from-config"
 
 
@@ -142,8 +142,8 @@ def test_regression_pr247_mcp_json_env_block(
     )
     cfg = Mem2MemConfig()
     load_config_overrides(cfg)
-    assert str(cfg.storage.sqlite_path) == "~/.memtomem/memtomem.db"
-    assert [str(p) for p in cfg.indexing.memory_dirs] == ["~/notes"]
+    assert Path(cfg.storage.sqlite_path).as_posix() == "~/.memtomem/memtomem.db"
+    assert [Path(p).as_posix() for p in cfg.indexing.memory_dirs] == ["~/notes"]
 
 
 # ---------------------------------------------------------------------------
@@ -221,7 +221,7 @@ def test_config_d_env_wins_over_fragments(
     )
     cfg = Mem2MemConfig()
     load_config_d(cfg)
-    assert str(cfg.storage.sqlite_path) == "/from/env.db"
+    assert Path(cfg.storage.sqlite_path).as_posix() == "/from/env.db"
 
 
 def test_config_d_unknown_section_warned_but_not_fatal(

--- a/packages/memtomem/tests/test_session_summary_rescue.py
+++ b/packages/memtomem/tests/test_session_summary_rescue.py
@@ -182,7 +182,7 @@ class TestBoostSourcesHelper:
 
         pipeline = _make_pipeline(storage, session_summary_config=cfg)
         sources = await pipeline._session_summary_boost_sources("q")
-        assert sources == {"/tmp/src/a.md", "/tmp/src/b.md"}
+        assert {Path(s).as_posix() for s in sources} == {"/tmp/src/a.md", "/tmp/src/b.md"}
         # Walk used the correct link_type
         call_args = storage.get_chunks_shared_from.await_args
         assert call_args.kwargs.get("link_type") == "summarizes"

--- a/packages/memtomem/tests/test_storage.py
+++ b/packages/memtomem/tests/test_storage.py
@@ -201,7 +201,10 @@ class TestNormPathUnicode:
 
     def test_norm_path_osError_fallback_still_normalizes(self, monkeypatch):
         # If ``Path.resolve`` raises, ``norm_path`` falls back to the input
-        # string — it must still NFC-normalize that fallback.
+        # string — it must still NFC-normalize that fallback. Comparison goes
+        # through ``as_posix()`` so the assertion is portable: norm_path uses
+        # ``str(p)`` on fallback which is platform-separator-dependent
+        # (backslash on Windows), but the NFC property under test is not.
         nfd = unicodedata.normalize("NFD", "/tmp/내 드라이브/file.md")
 
         def _boom(self, strict=False):
@@ -210,4 +213,4 @@ class TestNormPathUnicode:
         monkeypatch.setattr(Path, "resolve", _boom)
 
         out = norm_path(Path(nfd))
-        assert out == unicodedata.normalize("NFC", nfd)
+        assert Path(out).as_posix() == unicodedata.normalize("NFC", nfd)

